exchanges/credentials.go

```diff
@@ -115,31 +115,29 @@ func (b *Base) GetCredentials(ctx context.Context) (*account.Credentials, error)
 	if value != nil {
 		ctxCredStore, ok := value.(*account.ContextCredentialsStore)
 		if !ok {
-			// NOTE: Return empty credentials on error to limit panic on
-			// websocket handling.
-			return &account.Credentials{}, errContextCredentialsFailure
+			return nil, common.GetTypeAssertError("*account.ContextCredentialsStore", value)
 		}
 
 		creds := ctxCredStore.Get()
 		if err := b.CheckCredentials(creds, true); err != nil {
-			return creds, fmt.Errorf("context credentials issue: %w", err)
+			return nil, fmt.Errorf("context credentials issue: %w", err)
 		}
 		return creds, nil
 	}
 
+	// Fallback to exchange loaded credentials
+	b.API.credMu.RLock()
 	creds := b.API.credentials
-	err := b.CheckCredentials(&creds, false)
-	if err != nil {
-		// NOTE: Return empty credentials on error to limit panic on websocket
-		// handling.
-		return &account.Credentials{}, err
+	b.API.credMu.RUnlock()
+	if err := b.CheckCredentials(&creds, false); err != nil {
+		return nil, fmt.Errorf("invalid default credentials: %w", err)
 	}
-	subAccountOverride, ok := ctx.Value(account.ContextSubAccountFlag).(string)
-	b.API.credMu.RLock()
-	defer b.API.credMu.RUnlock()
-	if ok {
+
+	// Sub account override if set
```

Comment:
🐒 Nitpick: This comment adds no information we don't already have
Best rule is to use comments to explain either:
* Why are we doing something (as opposed to what are we doing)
* What or How we're doing something if it's too complicated to read immediately as code
In this case, the var name and `ok` give us everything we needed to know.
```suggestion
	// Sub account override if set
```